baseapp/baseapp.go

```diff
@@ -223,12 +223,21 @@ func (app *BaseApp) Name() string {
 	return app.name
 }
 
-func (app *BaseApp) InitAppVersion(ctx sdk.Context) {
-	if app.appVersion == 0 && app.paramStore.Has(ctx, ParamStoreKeyVersionParams) {
+// GetAppVersionFromParamStore returns the app version from the param store.
+func (app *BaseApp) GetAppVersionFromParamStore(ctx sdk.Context) uint64 {
+	if app.paramStore.Has(ctx, ParamStoreKeyVersionParams) {
 		var vp tmproto.VersionParams
 		app.paramStore.Get(ctx, ParamStoreKeyVersionParams, &vp)
-		// set the app version
-		app.appVersion = vp.AppVersion
+		return vp.AppVersion
+	}
+	return 0
```

Comment:
What is app version 0? Just a non-empty value to return?

Reply:
Yeah, it just means that it's not current set